base/abstractarray.jl

```diff
@@ -1632,6 +1632,14 @@ end
 
 typed_hcat(::Type{T}, A::AbstractVecOrMat...) where {T} = _typed_hcat(T, A)
 
+# Catch indexing errors like v[i +1] (instead of v[i+1] or v[i + 1]), where indexing is
+# interpreted as a typed concatenation. (issue #49676)
+typed_hcat(::AbstractArray, other...) = throw(ArgumentError("It is unclear whether you \
+    intend to perform an indexing operation or typed concatenation. If you intend to \
+    perform indexing (v[1 + 2]) insert missing operator or adjust spacing to clarify. \
```

Comment:
Just a minor suggestion: it'll be easier to read if there's a comma after `perform indexing (v[1 + 2])` and similarly after `typed concatenation (T[1 2])`. I'd also move `adjust spacing` before `insert missing operator`, as this is probably the common tripping hazard here.

Reply:
Makes sense. Updated.